src/Opaleye/Internal/Sql.hs

```diff
@@ -108,6 +118,7 @@ baseTable ti columns = SelectFrom $
               , tables = [Table (HSql.SqlTable (PQ.tiSchemaName ti) (PQ.tiTableName ti))] }
 
 product :: NEL.NonEmpty Select -> [HPQ.PrimExpr] -> Select
+product (t NEL.:| []) [] = t
```

Comment:
This looks like an un-related change, I'll revert that (and try and figure out where it came from!)